src/openrct2/ride/Vehicle.cpp

```diff
@@ -7218,15 +7217,12 @@ void Vehicle::UpdateAdditionalAnimation()
     {
         case VEHICLE_ENTRY_ANIMATION_MINITURE_RAILWAY_LOCOMOTIVE: // loc_6D652B
             *curVar_C8 += _vehicleVelocityF64E08;
-            al = (*curVar_C8 >> 20) & 3;
-            if (animation_frame != al)
-            {
-                ah = al;
-                al = animation_frame;
-                animation_frame = ah;
-                al &= 0x02;
-                ah &= 0x02;
-                if (al != ah)
+            currentFrame = ((*curVar_C8 * vehicleEntry->animation_speed_multiplier) >> vehicleEntry->animation_speed_exponent)
+                & 3;
+            if (animation_frame != currentFrame)
+            {
+                animation_frame = currentFrame;
+                if ((currentFrame & 0x01) == 0) // every even frame create a steam particle
```

Comment:
This is subtly different to the original. The original only shows new steam if the oldFrame and the newFrame have a mismatched second bit. This new version shows new steam if the newFrame is odd. Is this a conscious change that is being made or just for the sake of simplification. I would prefer we matched the original as much as possible but understand you might have made this change for a reason.